examples/options/src/details/configuration_printable.cpp

```diff
@@ -0,0 +1,25 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2023-2025 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Local include(s).
+#include "traccc/options/details/configuration_printable.hpp"
+
+// System include(s).
+#include <sstream>
+
+namespace traccc::opts {
+
+configuration_printable::~configuration_printable() = default;
+
+std::string configuration_printable::str() const {
+
+    std::ostringstream out;
+    print_impl(out, "", "", 0, get_max_key_width_impl());
+    return out.str();
+}
```

Comment:
This is not going to fly; when we move to using loggers here we will need to be able to print this configuration line-by-line, and producing a whole string first and then splitting it on newlines is not practical.